File: edit_image.py
from PIL import Image, ImageDraw
import os

def clean_portfolio_image(input_path, output_path):
    # Abrir la imagen original
    with Image.open(input_path) as img:
        # Coordenadas sugeridas con margen de seguridad (X: 1996, Y: 1561, W: 359, H: 70)
        # Se calculan respecto a la imagen original de 3834 x 2045
        rect_x, rect_y = 1996, 1561
        rect_w, rect_h = 359, 70
        
        # Obtener el color del fondo (gris de la interfaz de Grasshopper) 
        # Tomando una muestra de un píxel cercano al área de texto
        pixel_color = img.getpixel((2000, 1550)) # Un poco arriba del texto
        
        # Dibujar el rectángulo para ocultar el texto
        draw = ImageDraw.Draw(img)
        draw.rectangle([rect_x, rect_y, rect_x + rect_w, rect_y + rect_h], fill=pixel_color)
        
        # Guardar en WebP manteniendo la máxima calidad (100)
        img.save(output_path, "WEBP", quality=100, lossless=True)
        print(f"✅ Imagen guardada exitosamente en: {output_path}")

# Rutas
original_path = r"c:\Desarrollo\mmapp\mariomojica-portfolio\public\portfolio\2026_Mario_Mojica\Silla_Grasshopper\Silla_3D_Print_01.webp"
new_path = r"c:\Desarrollo\mmapp\mariomojica-portfolio\public\portfolio\2026_Mario_Mojica\Silla_Grasshopper\Silla_3D_Print_01_fixed.webp"

if __name__ == "__main__":
    if os.path.exists(original_path):
        clean_portfolio_image(original_path, new_path)
    else:
        print(f"❌ No se encontró la imagen en {original_path}")

File: edit_image.js
const sharp = require('sharp');
const fs = require('fs');

async function editImage() {
    const inputPath = 'c:/Desarrollo/mmapp/mariomojica-portfolio/public/portfolio/2026_Mario_Mojica/Silla_Grasshopper/Silla_3D_Print_01.webp';
    const outputPath = 'c:/Desarrollo/mmapp/mariomojica-portfolio/public/portfolio/2026_Mario_Mojica/Silla_Grasshopper/Silla_3D_Print_01_fixed.webp';

    try {
        // Obtenemos metadatos para asegurar las proporciones
        const metadata = await sharp(inputPath).metadata();
        
        // Coordenadas calculadas para el bloque de texto (X: 1996, Y: 1561, W: 359, H: 70)
        // en una imagen de 3834 x 2045.
        // El color es aproximadamente el gris de fondo de Grasshopper (#C9CBCD o similar)
        
        // Creamos un rectángulo de color sólido (gris interfaz)
        const svgRect = Buffer.from(
            `<svg width="359" height="70">
                <rect width="359" height="70" fill="#E8EBED" />
            </svg>`
        );

        await sharp(inputPath)
            .composite([{
                input: svgRect,
                top: 1561,
                left: 1996
            }])
            .webp({ quality: 100, lossless: true })
            .toFile(outputPath);

        console.log(`✅ Edición quirúrgica completada sin pérdida de calidad: ${outputPath}`);
    } catch (err) {
        console.error('❌ Error editando la imagen:', err);
    }
}

editImage();
